packages/profile/src/hooks/starknetid.ts

```diff
@@ -0,0 +1,24 @@
+import { useQuery } from "react-query";
+import { useConnection } from "./context";
+import { useState } from "react";
+
+export const useStarkAddress = ({ name }: { name: string }) => {
+  const { provider } = useConnection();
+  const [error, setError] = useState<string | null>(null);
+
+  const { data, isFetching } = useQuery({
```

Comment:
I added swc dep for async fetching logic but you make me realize that react query is just enough 👍